### PR TITLE
Getting existed and updating Interview request

### PR DESCRIPTION
**Pull request by ovoskresenskyy**

### CODE REVIEW

File: src/main/java/com/ratifire/devrate/repository/interview/InterviewRequestRepository.java [hunk 1]
```diff
@@ -50,4 +52,6 @@ List<InterviewRequest> findMatchedInterviewers(@Param("request") InterviewReques
       List<User> ignoreList);
 
   List<InterviewRequest> findByActiveTrueAndExpiredAtBefore(ZonedDateTime currentDateTime);
+
+  Optional<InterviewRequest> findByUserAndRole(User user, InterviewRequestRole role);
```
Comment (kartius) [inline src/main/java/com/ratifire/devrate/repository/interview/InterviewRequestRepository.java:56]: What if try to rename method findByUserIdAndRole(long userId, InterviewRequestRole role) ?

File: src/main/java/com/ratifire/devrate/service/interview/InterviewRequestService.java [hunk 2]
```diff
@@ -26,6 +27,19 @@ public InterviewRequest save(InterviewRequest interviewRequest) {
...
 
+  /**
+   * Finds an interview request for the given user and role. If not found, returns a new empty
+   * InterviewRequest.
+   *
+   * @param user the user associated with the interview request
+   * @param role the role of the interview request
+   * @return the found InterviewRequest or an empty InterviewRequest if not found
+   */
+  public InterviewRequest findByUserAndRole(User user, InterviewRequestRole role) {
```
Comment (kartius) [inline src/main/java/com/ratifire/devrate/service/interview/InterviewRequestService.java:38]: By the way, could we get List<InterviewRequest> ?

File: src/main/java/com/ratifire/devrate/service/interview/InterviewRequestService.java [hunk 3]
```diff
@@ -26,6 +27,19 @@ public InterviewRequest save(InterviewRequest interviewRequest) {
...
+    return interviewRequestRepository.findByUserAndRole(user, role)
+        .orElse(InterviewRequest.builder().build());
```
Comment (kartius) [inline src/main/java/com/ratifire/devrate/service/interview/InterviewRequestService.java:40]: Should it be empty InterviewRequest or just null?

File: src/main/java/com/ratifire/devrate/service/user/UserService.java [hunk 4]
```diff
@@ -612,6 +640,21 @@ private InterviewRequest createInterviewRequest(long userId, InterviewRequestDto
...
+   * Updates the interview request for the specified user based on the provided request DTO.
+   *
+   * @param userId     the user's ID
+   * @param requestDto the interview request details
+   * @return the updated InterviewRequest entity
+   */
+  private InterviewRequest updateInterviewRequest(long userId, InterviewRequestDto requestDto) {
+    User user = findUserById(userId);
+    InterviewRequest interviewRequest = interviewRequestService.findByUserAndRole(user,
+        requestDto.getRole());
```
Comment (kartius) [inline src/main/java/com/ratifire/devrate/service/user/UserService.java:653]: If we use findByUserIdAndRole we avoid double request to db

**Review by kartius — APPROVED** [review]

**Review by abalmasDA — APPROVED** [review]

**Pull request merged**